cdap-common/src/main/java/co/cask/cdap/common/kerberos/SecurityUtil.java

```diff
@@ -52,6 +55,11 @@
   private SecurityUtil() { }
 
   /**
+   * A pattern to match kerberos principals
+   */
+  private static final Pattern KERBEROS_PRINCIPAL = Pattern.compile("([^/@]*)(/([^/@]*))?@([^/@]*)");
```

Comment:
@chtyim I looked up that but then we will need to be aware of the default REALM. So, expectation was that the user will give with realm and I documented the supported type. Although, we can have support to detect default realm Hadoop's KerberosName does it already. Should we accept principal without realm then ?